nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -1423,9 +1423,40 @@ public static Timestamp toTimestamp(final Object value, final Supplier<DateForma
             }
         }
 
+        if ("oracle.sql.TIMESTAMP".equals(value.getClass().getName())) {
+            return toTimestampForOracle(value.toString(), fieldName);
+        }
+
         throw new IllegalTypeConversionException("Cannot convert value [" + value + "] of type " + value.getClass() + " to Timestamp for field " + fieldName);
     }
 
+    public static Timestamp toTimestampForOracle(String orcTimestamp, final String fieldName) {
+        // support Oracle TIMESTAMP, maybe loss accuracy
+        String orcFormat = "yyyy-MM-dd HH:mm:ss.SSS";
+        DateFormat dateFormat = getDateFormat(orcFormat);
```

Comment:
This could be optimized using a single static instance of `java.time.DateTimeFormatter` instead of recreating `DateFormat` for every method invocation. `DateTimeFormatter` can parse a string to an `Instant`, which contains the epoch date in milliseconds.